### ISSUE TITLE
Timeout not being honoured in Azure EventHub health check

### ISSUE THREAD

**Issue by josdeweger** [Xabaril/AspNetCore.Diagnostics.HealthChecks#1786]
**What happened**:
We are seeing lots of Unhealthy statuses coming back from the Azure EventHub check, which is configured with a timeout of 5 seconds, from the Aspnetcore.HealthChecks.AzureServiceBus package. We see the following exception message:
`Health check azureeventhub with status Unhealthy completed after 1001.0438ms with message '(null)'`

**What you expected to happen**:
The timeout that is configured during startup is being honoured

**How to reproduce it (as minimally and precisely as possible)**:
setup an app with the following registration:
```
services
    .AddAzureEventHub(
        connectionString: eventHubOptions.ConnectionString, 
        eventHubName: eventHubOptions.EventHubName, 
        tags: new[] { LivenessTag },
        timeout: TimeSpan.FromSeconds(5))
```

**Source code sample**:
see above

**Anything else we need to know?**:

**Environment**:
- .NET Core version: .NET 6
- Healthchecks version: 6.1.0
- Operative system: Azure Container Apps

**Comment by sungam3r** [comment]
PR is welcome.

**Comment by sungam3r** [comment]
fixed in #1992
